### PR TITLE
MDEV-1: Add trap entity type to pacmossi project

### PR DESCRIPTION
**Pull request by chris-pollard**
### Changes:

Migrations:
- Add enum type 'trap' and add to entity model.
- Add 'trap' to canonical hierarchy in entity_hierarchy table.

### CODE REVIEW

File: packages/database/src/modelClasses/Entity.js [hunk 1]
```diff
@@ -66,6 +67,7 @@ const ENTITY_TYPES = {
   MEDICAL_AREA,
   NURSING_ZONE,
   FETP_GRADUATE,
+  TRAP,
```
Comment (billli0) [inline packages/database/src/modelClasses/Entity.js:70]: Adding new entity type here is not required now. See note: https://github.com/beyondessential/tupaia/blob/867d568f61759c3c524ff039c026f467d6cc6895/packages/database/src/modelClasses/Entity.js#L13

**Review by billli0 — CHANGES_REQUESTED** [review]
@chris-pollard, there are some updates on adding entity type. PR: https://github.com/beyondessential/tupaia/pull/4200

With that, adding new type can be done by executing `alter type` query without any code changes, if new type is successfully added, it will be on `Entity Type` section in admin panel. 

Minor changes like adding canonical types on `entity_hierarchy` table, feel free to do it via DBeaver or Postico. 

Better approach would be to peer with a dev and do it on dev first, but it is up to you if you want to merge this PR to insert new type without any more work.

File: packages/database/src/modelClasses/Entity.js (hunk 1 above)
Comment (chris-pollard) [reply packages/database/src/modelClasses/Entity.js:70]: Ah thanks for pointing that out @billli0 . Will update the PR. I might keep in the canonical type for this one anyway.

**Comment by chris-pollard** [comment]
Ok, that should be good now, thanks! @billli0

**Review by billli0 — APPROVED** [review]

**Comment by chris-pollard** [comment]
Added in production db manually

**Pull request closed without merging**